src/network/networkctl.c

```diff
@@ -2943,6 +2943,24 @@ static int verb_persistent_storage(int argc, char *argv[], void *userdata) {
         if (r < 0)
                 return r;
 
+        r = varlink_set_allow_fd_passing_output(vl, true);
+        if (r < 0)
+                return log_error_errno(r, "Failed to allow passing file descriptor through varlink: %m");
+
+        if (ready) {
+                _cleanup_close_ int fd = -EBADF;
+
+                fd = open("/var/lib/systemd/network/", O_CLOEXEC | O_DIRECTORY | O_PATH);
```

Comment:
Hmm, I wonder if it's better to use a non-O_PATH fd here, and verify the fd flags through `fd_verify_safe_flags` on server side

Reply:
Checking fd with `fd_verify_safe_flags()` sounds nice. But why we need to use non-O_PATH fd here? Does the check only work with non-O_PATH fd?

Reply:
The idea (as described in the comment of fd_verify_safe_flags) is to ensure the process that originally opened the fd has the permission to operate on it, rather than bypassing the permission check by sending an O_PATH fd over.